src/Chapter3.hs

```diff
@@ -971,6 +1104,28 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Week =
+    Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Eq,Ord,Enum,Show)
+
+isWeekend :: Week -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: Week -> Week
+nextDay Sunday = Monday
+nextDay w = toEnum (fromEnum w + 1)
```

Comment:
Unfortunately, `toEnum` will fail in the runtime when you go out the number of constructors. You would need to use `mod` here as well 🙂 
Alternatively, you could use the `Bounded` instance.